### PR TITLE
Update variables

### PR DESCRIPTION
**Pull request by msargen**
#14 

- Change the object variable names to be `<object name>_<variable_name>`
- Renamed some objects for ease of use
- Added some code to make how the code works a little more consistent (adding a create event to the level_end object)
- Cleaned up some of the code by adding spacing or parenthesis
- Added the `.yyp` file to the source control
- Updated the `.gitignore` to not reference the `.yyp`
- Update script function calls to be more dev friendly.

### Testing:
I didn't do a super thorough playthrough after finishing my changes, but I did test each change as I made them and they all still worked for normal use cases.

### CODE REVIEW

File: objects/obj_menu/Create_0.gml [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 menu_itemheight = font_get_size(fnt_menu);
 menu_committed = -1;
 menu_control = true;
 
-menu[4] = "Level 3";
-menu[3] = "Level 2";
-menu[2] = "Level 1";
-menu[1] = "Continue";
-menu[0] = "Quit";
+menu_menu[4] = "Level 3";
```
Comment (Scheills) [inline objects/obj_menu/Create_0.gml:17]: should these be menu_option instead of menu_menu?

File: objects/obj_player/Step_0.gml [hunk 2]
```diff
@@ -1,159 +1,159 @@
...
-	var _dir = key_right - key_left;
-	hsp += _dir * hsp_acc;
+	var _dir = player_key_right - player_key_left;
+	player_horizontal_speed += _dir * player_horizontal_speed_acc;
 	if (_dir == 0)
 	{
-		var _hsp_fric_final = hsp_fric_ground;
-		if (!onground) _hsp_fric_final = hsp_fric_air;
-		hsp = scr_approach(hsp,0,_hsp_fric_final);
+		var _horizontal_speed_fric_final = player_horizontal_speed_fric_ground;
```
Comment (Scheills) [inline objects/obj_player/Step_0.gml:26]: is there a reason _horizontal_speed_fric_final does not have a leading word?

File: objects/obj_player/Step_0.gml [hunk 3]
```diff
@@ -1,159 +1,159 @@
...
 }
 
 #endregion
 
 #region //Calculate Vertical Movement
-var _grv_final = grv;
-var _vsp_max_final = vsp_max;
-if (onwall != 0) && (vsp > 0)
+var _grv_final = player_gravity;
+var _vertical_speed_max_final = player_vertical_speed_max;
```
Comment (Scheills) [inline objects/obj_player/Step_0.gml:48]: Is there a reason _vertical_speed_max_final doesn't have a leading word?

File: objects/obj_player/Step_0.gml [hunk 4]
```diff
@@ -1,159 +1,159 @@
...
+if ((player_on_wall != 0) && (player_vertical_speed > 0))
 {
-	_grv_final = grv_wall;
-	_vsp_max_final = vsp_max_wall
+	_grv_final = player_gravity_wall;
```
Comment (Scheills) [inline objects/obj_player/Step_0.gml:51]: Another variable with no leading word

Comment (msargen) [reply objects/obj_player/Step_0.gml:51]: Those are local variables and the beginning `_` is intentional and a naming convention

File: objects/obj_menu/Create_0.gml (hunk 1 above)
Comment (msargen) [reply objects/obj_menu/Create_0.gml]: Menu object has been changed to main_menu

**Review by Scheills — APPROVED** [review]
Looks great! Thanks for explaining the variables that start with _

**Pull request merged**